discovery/generator.go

```diff
@@ -0,0 +1,52 @@
+package discovery
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func incString(s string) string {
+	if s == "" {
```

Comment:
changed to `switch` and I summarise what this code does as comments, case by case.